corehq/apps/linked_domain/updates.py

```diff
@@ -333,6 +334,10 @@ def update_data_dictionary(domain_link):
             case_property_obj.deprecated = case_property_desc['deprecated']
             case_property_obj.data_type = case_property_desc['data_type']
             case_property_obj.group = case_property_desc['group']
+            case_property_obj.group_obj, created = CasePropertyGroup.objects.get_or_create(
+                name=case_property_desc['group'],
+                case_type=case_type_obj,
+            )
```

Comment:
It looks like there are tests for the other data dictionary models in linked domains - could you add groups to that?  Also, I think these `CasePropertyGroup`s should be created from the upstream group objects, not the single string field, as that doesn't contain description or index information.

Reply:
Thanks, though I think there are a couple edge cases that need to be taken into account.  Group is optional, and by default, properties aren't given a group.  I believe this code will result in a blank group being created in the downstream domain for those properties.

On that note, you could also take the approach of making group mandatory, with a default "uncategorized" group for those where a group wasn't assigned explicitly.  I don't feel strongly one way or the other on that topic.

The other consideration is case property groups with no properties assigned.  I'm not sure whether these are allowed in the UI, but it's worth mentioning.  This code goes through every property and fetches or creates its group, but the reverse might simpler and more performant.  The code already iterates over case types, then properties within each type.  It'd make sense to me to insert property groups in between those two (including a null group for unassigned properties).  This would make the return value from `get_data_dictionary` look something like:
```python
{
    "case_type_1": {
        ...
        "groups": {
            None: {
                "properties": {
                    "property_1": {...}
                }
            },
            "group 1": {
                "description": "first group",
                "index": 1,
                "properties": {
                    "property_2": {...}
                }
            }
        }
    },
    "case_type_2": {...}
}
```
This way you wouldn't need to duplicate the information for each group inside every case property in the group, or `get_or_create` it each time.

Reply:
Thanks @esoergel for the info.  I have made the changes in these commits. The properties which don't have a group are grouped under `""` (empty string).

[b5139b8](https://github.com/dimagi/commcare-hq/pull/32821/commits/b5139b8c1a155f9366229ae851d115e830c8a469)
[cf51344](https://github.com/dimagi/commcare-hq/pull/32821/commits/cf513442712eed5e1cb2191cc7d56c0eceef37e6)
[af0f37b](https://github.com/dimagi/commcare-hq/pull/32821/commits/af0f37beac5b9c0389f53a583239c3c60132d1e6)